### ISSUE TITLE
onlyoffice-documentserver-de depends on ttf-mscorefonts-installer

### ISSUE THREAD

**Issue by hahaxd35** [ONLYOFFICE/DocumentServer#1743]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
while updating i was not able too update onlyoffice because "ttf-mscorefonts-installer" was missing and not able to install. "constib" was missing in /etc/apt/sources.list

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
apt upgrade

**What is the expected behavior?**
make "ttf-mscorefonts-installer" recommend not depends

**Did this work in previous versions of DocumentServer?**
yes

**DocumentServer version:**
7.1

**Operating System:**
Debian 10

**Comment by ShockwaveNN** [comment]
Hi, we added this dependency to allow better compatibility with MS Office files, a lot of them use fonts from MS so they are not rendered perfectly 

```ttf-mscorefonts-installer` is a part of [Debian contrib ](https://packages.debian.org/search?keywords=ttf-mscorefonts-installer) - you can enable this repo and install that package 

In the meantime we got issue 57135 in our private issue tracker to gather feedback from users and maybe revert it back, so thank you for your feedback

**Comment by jiriks74** [comment]
@hahaxd35 
You have 3 options here:
1. Install the `ttf-mscorefonts-installer`
2. Modify the deb package so that it's not a dependency
3. Use a docker (I'm currently maintaining one [on my GitHub repository](https://github.com/jiriks74/Docker-DocumentServer) you can find a guide on how to set it up there, it's really easy.

**Comment by hahaxd35** [comment]
>  "constib" was missing in /etc/apt/sources.list

thanks?!?

**Comment by ShockwaveNN** [comment]
> "constib" was missing in /etc/apt/sources.list

It was not missing, it's not enabled by default as far as I'm aware

But it's on Debian, on Ubuntu this package is included in default repo

**Comment by jiriks74** [comment]
@ShockwaveNN It would be nice if there'd be an package for no fonts. In docker I mount my own Microsoft fonts (I mount them directly from my Windows 10 partition) so I have newer and better looking ones than the ones provided by `ttf-mscorefonts-installer` as they're pretty old (according to ArchWiki the ones we are able redistribute are really old). I think that another package is the better solution as most of the people don't use fonts from a Windows installation and for those who do use their own fonts (or really don't want to use/have Microsoft fonts) there would be a package on your site (like `-no_mscore_fonts.deb`.

For regular people the compatibility is valid and I don't think that you should remove the dependency. But it does not allow us to use our own ms fonts without rebuilding the package so I think that having a package with that "tag" would be the best solution. What do you think?

**Comment by jiriks74** [comment]
I think that you have to keep in mind that there are many people that don't want Microsoft on their box at all and are using ONLYOFFICE only (or chose it mainly) because they're forced to use Microsoft OpenXML format's at school/work.

**Comment by ShockwaveNN** [comment]
@jiriks74 Ok thanks for you opinion, not sure that we'll able to add another package without dependency (it will produce to much chaos, since we already got 3 version of our DocumentServer package only for Debian-based system) but I add your feedback to our internal discussion

**Comment by jiriks74** [comment]
> @jiriks74 Ok thanks for you opinion, not sure that we'll able to add another package without dependency (it will produce to much chaos, since we already got 3 version of our DocumentServer package only for Debian-based system) but I add your feedback to our internal discussion

Oh, I forgot you have `x86` version as well. Yeah there would be a small chaos. Maybe, after my graduation, I'll learn how to make a repository and distribute modified packages (as I'd use them personally, why wouldn't I allow others)a

**Comment by hahaxd35** [comment]
nevermind, 

as far as i know constriib is not enabled by default in ubuntu, you have to select it on install 
https://wiki.ubuntuusers.de/sources.list/

Administrators maybe dont want MS on there Linux server, but customers wants... but required is not good

**Comment by jiriks74** [comment]
> nevermind,
> 
> as far as i know constriib is not enabled by default in ubuntu, you have to select it on install https://wiki.ubuntuusers.de/sources.list/
> 
> Administrators maybe dont want MS on there Linux server, but customers wants... but required is not good

You don't need contrib in ubuntu by default to get `ttf-mscorefonts-installer`. I used Ubuntu when 18.04 was maybe a year old and even back then it was included in the main repos. In the later versions it's there as well.

**Comment by jiriks74** [comment]
@hahaxd35 
I know, but you have to take a look at their perspective as well. This is for user experience. Imagine a person coming to Linux. They're using Mint, Ubuntu or whatever, and they've been recommended ONLYOFFICE as they currently have the best OpenXML (`.docx`, `.pptx`, `.xlsx`) compatibility. But oh, they cannot find their favorite fonts and the fonts that are used at work/school/etc. For a newbie it might be pretty hard to find how to install those fonts on their system.

So it's a loose loose situation. You gain better user experience for making some people angry OR you don't make anyone angry but you don't have better user experience. If I was in their place, I would probably choose the same as their product is mainly promoted as "Microsoft compatible" so it wouldn't really make sense if you wouldn't be compatible with their fonts.

**Comment by hahaxd35** [comment]
BTWI:
Debian default install recommended packages on its self, you have to disable "install recommend"

**Comment by ShockwaveNN** [comment]
I'll leave this issue open to be able to gather more feedback

**Comment by agentl074** [comment]
I use LibreOffice -- which uses Liberation Fonts as the default, and they work, but I did install the corefonts package since I want the most compatibility with Word docs. I did experiment with using a combination of Times and Liberation Serif, and while at first glance, they do look similar, I find the old newspaper font (Times New Roman), a little better for me. Arial isn't bad either -- but if you're writing a long article on print, I still think Times is the best. 

Bottom line, LibreOffice works perfectly for all of my needs -- and I also use Google Docs. Google Docs is actually the best.  Most Mac users prefer Google Docs.

**Comment by jiriks74** [comment]
@hahaxd35
But take into account that people like me use this software as well. When I started out I had no idea how the servers work and I just needed something that would easily allow me to edit my schools documents.  When people are starting out with documents it can be really helpful when it set's up everything it says it does.
It advertises really good compatibility with MS Office and you break that by not providing fonts that 90% of documents created with it use.

I understand the point of not using the fonts, but I could say that when you're at the level where you care about these things you can just make a simple script of 4 commands to remove the dependency. Unpack, remove the package dependency, repack, install. Especially if it's for corporate, it's just easy to make something like this happen. What I would do:

1. Hold the package in APT
2. Create the script + add a check for update (so you won't rebuild the package when there's no update)
3. Add it as a cron job.

And you're done.

And from what I understand, it was not really made for private use. It was closed-source collaboration software that they decided to open up for the FOSS community so that they could basically enjoy something nice. 90% from that community won't be paying anything. And the percentage of those contributing won't be big as well. Most of us are just adding bugs for them to figure out.

**Comment by hahaxd35** [comment]
if you dont have any idea how to start, why not using docker? its designt for... there can be images with 100% m$ compatibility thats not my point, 

Imagine Apache doing something like this for newbies xD but I don't care now. 



i have already many scripts, (that need time too) that modify onlyoffice my way, but on any update i have to spent time.
this is for software we pay for.... 

but not a discussion on github
